### PR TITLE
Always return a symbol from compute_serializer_name.

### PR DESCRIPTION
**Pull request by carsol**
I made a previous change that if you passed a class instead of a symbol, it would just return it (https://github.com/Netflix/fast_jsonapi/pull/198). This method works either way but it makes more sense to be uniform about what we return (always a symbol class). It also allows for us to clean up a test also.

### Related PRs:
#198
#201

---
This change is [Reviewable](https://reviewable.io/reviews/netflix/fast_jsonapi/206)

### CODE REVIEW

**Comment by carsol** [comment]
Related PRs: 
- https://github.com/Netflix/fast_jsonapi/pull/198
- https://github.com/Netflix/fast_jsonapi/pull/201

File: lib/fast_jsonapi/object_serializer.rb [hunk 1]
```diff
@@ -221,7 +221,7 @@ def create_relationship_hash(base_key, relationship_type, options, block)
       end
 
       def compute_serializer_name(serializer_key)
-        return serializer_key unless serializer_key.is_a? Symbol
+        return serializer_key.to_s.to_sym unless serializer_key.is_a? Symbol
```
Comment (zinosama) [inline lib/fast_jsonapi/object_serializer.rb]: The output of this method is always used as
`@relationships_to_serialize[item][:serializer].to_s.constantize` 

so why not change this method to simply return a string? Not sure what's the point of converting to symbol then back to string.

Comment (carsol) [reply lib/fast_jsonapi/object_serializer.rb]: Good point. I think it would just make sense always return a `constantize` serializer Class instead of a String or Symbol. It cuts the conversation.

So this would look like this:
```
      def compute_serializer_name(serializer_key)
        return serializer_key.to_s.constantize unless serializer_key.is_a? Symbol
        namespace = self.name.gsub(/()?\w+Serializer$/, '')
        serializer_name = serializer_key.to_s.classify + 'Serializer'
        return (namespace + serializer_name).constantize if namespace.present?
        (serializer_key.to_s.classify + 'Serializer').constantize
      end
```

By doing this we can make the output be:
```
@relationships_to_serialize[item][:serializer]
```

Thoughts @zinosama ?

Comment (zinosama) [reply lib/fast_jsonapi/object_serializer.rb]: If the serializer class doesn't exist, this approach will display a more accurate/upstream source of error - thus I think it's a better solution.

File: spec/lib/object_serializer_class_methods_spec.rb [hunk 2]
```diff
@@ -251,7 +251,7 @@
           attributes :name, :release_year
           has_many :actors
           belongs_to :owner, record_type: :user
-          belongs_to :movie_type, serializer: "#{key_transform}_movie_type".to_sym
+          belongs_to :movie_type, serializer: "#{key_transform}_movie_type"
```
Comment (zinosama) [inline spec/lib/object_serializer_class_methods_spec.rb]: Also, i'm not sure what's the reason for this change - it seems to fail the spec 😞
